var/spack/repos/builtin/packages/heffte/package.py

```diff
@@ -146,17 +126,19 @@ def test_make_test(self):
         cmake_dir = self.test_suite.current_test_cache_dir.testing
 
         options = [cmake_dir]
+        options.append(f"-DHeffte_DIR={self.spec.prefix.lib.cmake.Heffte}")
         if "+rocm" in self.spec:
+            # path name is 'hsa-runtime64' but python cannot have '-' in variable name
+            hsa_runtime = join_path(self.spec["hsa-rocr-dev"].prefix.lib.cmake, "hsa-runtime64")
             options.extend(
                 [
                     f"-Dhip_DIR={self.spec['hip'].prefix.lib.cmake.hip}",
                     "-DAMDDeviceLibs_DIR="
                     + f"{self.spec['llvm-amdgpu'].prefix.lib.cmake.AMDDeviceLibs}",
                     f"-Damd_comgr_DIR={self.spec['comgr'].prefix.lib.cmake.amd_comgr}",
```

Comment:
These should also be updated to use `self.define(key, value)` syntax.

Reply:
I updated the syntax